open-apparel-registry/open-apparel-registry
# Automate importing SourceMap data dumps

## Overview

* Create local volumes for ElasticSearch and MongoDB data
* Add `scripts/bootstrap` which pulls down environment files and imports SourceMap data dumps
* Add `scripts/generate_fixed_dumps` which generates new database dumps that include an API key for the Firebase master account specified in `.env.backend`
* Add `create_master_account_api_key.js` which creates a new API key for the Firebase master account
* Add `synchronize.js` which allows running [ElasticSearch indexing](https://github.com/mongoosastic/mongoosastic#indexing-an-existing-collection) outside of the web application
* Read in Firebase configuration from environment variables

Resolves #4 and #9 

## Demo

<img width="1735" alt="image" src="https://user-images.githubusercontent.com/1774125/50029482-e07ea500-ffc0-11e8-89ab-5cca353223ac.png">

<img width="1735" alt="image" src="https://user-images.githubusercontent.com/1774125/50029534-0310be00-ffc1-11e8-9b26-8c87789a4a37.png">

## Notes

In response to Firebase warnings in my browser's console, I had to into the Firebase console and create a "Realtime Database" for our project. Then, I had to modify the database rules:

```diff
 {
   /* Visit https://firebase.google.com/docs/database/security to learn more about security rules. */
   "rules": {
-    ".read": false,
-    ".write": false
+    ".read": true,
+    ".write": true
   }
 }
```

## Testing Instructions

Bring up a fresh development environment:

```bash
$ export AWS_PROFILE=open-apparel-registry
$ ./scripts/setup
```

Visit https://localhost:3000 and see if you can register an account and search through the existing data.
